src/citrine/ara/variables.py

```diff
@@ -0,0 +1,98 @@
+"""Variable definitions for Ara."""
+from typing import Type, Optional, List  # noqa: F401
+from abc import abstractmethod
+
+from taurus.entity.link_by_uid import LinkByUID
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization import properties
+
+
+class Variable(PolymorphicSerializable['Variable']):
+    """A variable that can be assigned values present in material histories.
```

Comment:
Shouldn't the `output_name` be defined for all `Variable` classes and thus have an abstract method at the parent class level?  Also, this feels more like a taxonomy or hierarchy of labeling, so `output_name` seems to miss on the plurality of terms and their nested nature.  Not that I have a better word at the moment.

Reply:
I cannot easily move them into the parent class (see https://github.com/CitrineInformatics/citrine-python/issues/166 ).  Agreed that its the ideal way to do it, but we don't have members like that defined in any of our parent classes.